components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/dao/impl/APIResourceManagementDAOImpl.java

```diff
@@ -783,15 +828,42 @@ private static APIResource getApiResource(ResultSet resultSet, List<APIResourceP
      * @return SQL statement to retrieve API resources.
      */
     private String buildGetAPIResourcesSqlStatement(String databaseName, Integer tenantId, String filterQuery,
-                                                    String sortOrder, Integer limit) {
+                                                    String sortOrder, Integer limit) throws APIResourceMgtException {
 
         String sqlStmtHead = SQLConstants.GET_API_RESOURCES;
         String sqlStmtTail = SQLConstants.GET_API_RESOURCES_TAIL;
+        int initialTenantId = -1234;
```

Comment:
Why -1234 is used here?

If it's a valid reason let's use https://github.com/yasassri/carbon4-kernel/blob/a1f3802212a02a939f07641608620dc03affab8f/core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/multitenancy/MultitenantConstants.java#L33

Reply:
by hard coding this, it seems other root orgs have an impact

Reply:
Used the constant. Here the new variable is used only for the validation. So using the super tenant tenant id will not be an impact on the root orgs.